File: src/untrusted/nacl/sleep.c
/*
 * Copyright 2009 The Native Client Authors.  All rights reserved.
 * Use of this source code is governed by a BSD-style license that can
 * be found in the LICENSE file.
 */

#include <sys/types.h>
#include <sys/time.h>
#include <sys/nacl_syscalls.h>

unsigned int sleep(unsigned int seconds) {
  struct timespec req;
  struct timespec rem;

  req.tv_sec = seconds;
  req.tv_nsec = 0;

  if (-1 == nanosleep(&req, &rem)) {
    return rem.tv_sec;
  }
  return 0;
}

File: tools/isolate/data/run_test_cases/sleep.py
#!/usr/bin/env python
# Copyright (c) 2012 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

"""Sleeps."""

import sys
import time


def main():
  assert len(sys.argv) == 2
  print 'Sleeping.'
  sys.stdout.flush()
  time.sleep(float(sys.argv[1]))
  print 'Slept.'
  return 0


if __name__ == '__main__':
  sys.exit(main())
